docs/spec/consensus/light-client.md

```diff
@@ -1,113 +1,316 @@
+
+```go
+  func CheckSupport(h1,h2,trustlevel,trustlevel_adj) bool {
+    if h1.Header.bfttime + tp < now { // Observation 1
+      return false // old header was once trusted but it is expired
+    }  
+    vp_all := totalVotingPower(h1.Header.NextV)
+      // total sum of voting power of validators in h2
+
+    if h2.Header.height == h1.Header.height + 1 {
+      // specific check for adjacent headers; everything must be
+      // properly signed.
+      // also check that h2.Header.V == h1.Header.NextV
+      // Plus the following check that 2/3 of the voting power
+      // in h1 signed h2
+      return (votingpower_in(signers(h2.Commit),h1.Header.NextV) >
+              max(2/3,trustlevel_adj) * vp_all)
+        // signing validators are more than two third in h1.
+        // headers in Cosmos are typically signed by all validators, so one
+        // may request that all signed (liveness might be endangered)
+    }
+
+    s := intersection(h1.Header.NextV,signers(h2.Commit))
+      // get validators in h1 that signed h2
+    vps := votingpower_in(s,h1.Header.NextV)
+      // sum of voting powers in h1 of
+      // validators that signed h2
+    return (vps > max(1/3,trustlevel) * vp_all)
+      // signing validators are more than a third in h1
+  }
+
+  *Remark*: Basic header verification must be done for *h2*. Similar checks are done in:  
+  https://github.com/tendermint/tendermint/blob/master/types/validator_set.go#L591-L633
+
+  *Remark*: There are some sanity checks which are not in the code:
+  *h2.Header.height > h1.Header.height* and *h2.Header.bfttime > h1.Header.bfttime* and *h2.Header.bfttime < now*.
+
+
+
+*Correctness arguments*
+
+Towards Lite Client Accuracy:
+- Assume by contradiction that *h2* was not generated correctly and the lite client sets trust to true because *CheckSupport* returns true.
+- h1 is trusted and sufficiently new
+- by Tendermint Fault Model, less than 1/3 of voting power held by faulty validators => at least one correct validator *v* has signed *h2*.
+- as *v* is correct up to now, it followed the Tendermint consensus protocol at least up to signing *h2* => *h2* was correctly generated, we arrive at the required contradiction.
+
+
+Towards Lite Client Completeness:
+- The check is successful if sufficiently many validators of *h1* are still validators in *h2* and signed *h2*.
+- If *h2.Header.height = h1.Header.height + 1*, and both headers were generated correctly, the test passes
+
+*Verification Condition:* We may need a Tendermint invariant stating that if *h2.Header.height = h1.Header.height + 1* then *signers(h2.Commit) \subseteq h1.Header.NextV*.
+
+*Remark*: The variable *trustlevel* can be used if we believe that relying on one correct validator is not sufficient. However, in case of (frequent) changes in the validator set, the higher the *trustlevel* is chosen, the more unlikely it becomes that CheckSupport returns true for non-adjacent headers.
```

Comment:
> f we believe that relying on one correct validator is not sufficient

who are we here? if relying on one correct validator is enough from security standpoint, then we don't need `trustlevel`

Reply:
I tried to clarify that "we" is the user in the new revision...

Reply:
> Does that make sense?

Makes sense, thanks!